### PR TITLE
chore(TS): add typescript definitions

### PR DESCRIPTION
**Pull request by TylorS**

### CODE REVIEW

File: type-definitions/prelude.d.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+export function findIndex<A>(item: A, arr: A[]): number;
+
+export function isArrayLike(x: any): boolean;
+
+// function.js
+
+export function id<A>(x: A): A;
+
+export function compose<A, B, C>(f: (val: C) => C, g: (val: A) => C): (val: A) => C;
```
Comment (briancavalier) [inline type-definitions/prelude.d.ts]: B is missing:

``` js
export function compose<A, B, C>(f: (val: B) => C, g: (val: A) => B): (val: A) => C;
```

File: type-definitions/prelude.d.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export function apply<A, C>(f: (val: A) => C, x: A): C;
```
Comment (briancavalier) [inline type-definitions/prelude.d.ts]: Let's use A and B instead of A and C.

File: type-definitions/prelude.d.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  (a: A): (b: B) => C;
+  (a: A, b: B): C;
+}
+
+interface CurriedFunction3<A, B, C, D> {
+  (): CurriedFunction3<A, B, C, D>;
+  (a: A): CurriedFunction2<B, C, D>;
+  (a: A, b: B): (c: C) => D;
+  (a: A, b: B, c: C): D;
+}
```
Comment (briancavalier) [inline type-definitions/prelude.d.ts:45]: These are clever!  Nicely done.

File: type-definitions/prelude.d.ts (hunk 1 above)
Comment (TylorS) [inline type-definitions/prelude.d.ts]: Good catch

File: type-definitions/prelude.d.ts (hunk 2 above)
Comment (TylorS) [inline type-definitions/prelude.d.ts]: Another good catch! :+1:

File: type-definitions/prelude.d.ts (hunk 3 above)
Comment (TylorS) [inline type-definitions/prelude.d.ts:45]: This is where TypeScript sucks

**Comment by TylorS** [comment]
Fixed!

**Comment by briancavalier** [comment]
LGTM

**Comment by TylorS** [comment]
LGTM

**Pull request merged**